### ISSUE TITLE
Failed to update config file on windows

### ISSUE THREAD

**Issue by matthiasng** [profclems/glab#210]
Running `glab config -t ...` failes with `Failed to update config file: chmod C:\Users\...\AppData\Local\Temp\..env337738213: not supported by windows`

**Description**
https://github.com/profclems/glab/blob/3ff88b6ea5bb8e2aba83d6e29f9cefac533a0979/internal/config/config.go#L162
renameio.WriteFile uses [File.Chmod](https://golang.org/pkg/os/#File.Chmod) which is not supported on windows.

**Expected Behavior**
`glab config` command can write the config file

**Actual Behavior**
`glab config` failes when writing the config file

**Possible Fix**
Remove renameio and add some error handling to the open-write go code.
I think that should be enough to guarantee (99.99... %) not to destroy the config file.
@andhe please correct me if i'm wrong.
Alternative, fix https://github.com/google/renameio/issues/17

**Steps to Reproduce**
1. run `glab config -t 123`

**Your Environment**
* Version used: Branch:trunk
* Operating System and version: Windows 10

**Comment by andhe** [comment]
Ouch, sorry for this problem.

Maybe we can fix renameio to work on Windows too though. /cc @stapelberg
(Edit: oh you already linked the relevant renameio issue...)

**Comment by stapelberg** [comment]
I’d love to receive a fix for this in renameio, but have no idea about windows myself :)

**Comment by profclems** [comment]
Atomic file replacement in windows seems almost impossible unless with hardcoded routines

**Comment by andhe** [comment]
@matthiasng I'm not sure how to do it, but is there any chance you could hack your renameio, replacing `t.Chmod(perm)` in writefile.go with `os.Chmod(t.Name(), perm)`, and see if that resolves the issue? (I'm not sure if the chmod is the only issue with renameio on windows...)

**Comment by profclems** [comment]
We can write a temporary fix for glab by eliminating the `chmod` function on windows till the issue is resolved in renameio:

https://github.com/google/renameio/blob/353f8196982447d8b12c64f69530e657331e3dbc/writefile.go#L29

with:
```go
if runtime.GOOS != "windows" {
	if err := t.Chmod(perm); err != nil {
		return err
	}
}
```

**Comment by andhe** [comment]
I've submitted https://github.com/google/renameio/pull/21 which was tested with a trivial test program importing my forked version from `github.com/andhe/renameio`.

**Comment by matthiasng** [comment]
Following your PR in renameio it seems that this library will completely drop windows support, which in my opinion is right desigion because there is no reliable way to gurantee a atomic rename on windows.

I understand that with #170 the config loading and saving is getting to replaces by viper, am i right ? Therefore, i think that a simple workaround for windows is good enought at the moment.

**Comment by andhe** [comment]
FYI example at https://github.com/google/renameio/pull/21#issuecomment-687920335 - the switch statement could just be dropped in place of the only use of renameio.WriteFile that currently exists in glab trunk.

I'm thinking the need of importing/linking `runtime` (and `renameio` on windows) could be avoided by creating a `WriteFile` helper method in afile which has a `!+build windows` and another `writefile_windows.go` which implements `WrifeFile` using `ioutil.WriteFile`.

Since I feel somewhat responsibel for creating this problem I'm willing to spend some more time on fixing things up. Please let me know if this is something you want and in which direction you'd like to see the implementation go.

(Also just simply switching to ioutil.WriteFile would still be an improvement over the previous code that was replaced with renameio.WriteFile. Which might be the most straight forward solution to avoid spending time on this regression if there are other more urgent things to adress to get a new release out the door.)

**Comment by matthiasng** [comment]
I think renameio only fixes the sympton and not the root cause. There are several variables in the configuration, and for each of them glab reads and write the configration separately. I think reading and writing the entire file only once for all variable should be good enough (not in scope of this issue but an implicite upvote for #170).

I have also never seen a cli tool that uses atomic write operations for configfiles

Therefore, I think that it is not worth investing a lot of time because a simple ioutil. WriteFile should be enough.

**Comment by stapelberg** [comment]
> I have also never seen a cli tool that uses atomic write operations for configfiles
> 
> Therefore, I think that it is not worth investing a lot of time because a simple ioutil. WriteFile should be enough.

If you’re okay with emptying out your config file when you run out of disk space,
or losing your config file contents when your laptop battery runs out while you use the tool, sure.

Personally, I think it’s better to avoid these cases, so I recommend using renameio.WriteFile where available.
